session.go

```diff
@@ -850,6 +847,12 @@ func (s *Session) deleteStream(id uint32) {
 		return
 	}
 	s.memoryManager.ReleaseMemory(str.memory)
+	if s.client == (id%2 == 0) {
+		if s.numIncomingStreams == 0 {
+			panic("yamux BUG: numIncomingStream would be negative")
```

Comment:
This panic is not caught by our panic handlers if it was caused by the application calling `Close` or `Reset` on the stream. Not really sure what the best thing to do here is. We want to surface this bug, but we can't use the normal logger for that. Every user of this library will have logging turned off, because the amount of things we log is just crazy.

Reply:
lets just ERROR log BUG very loudly, panic is innapropriate for our our applications.

Reply:
we should just change the logging to use go-log, it is very much forked already.

Reply:
Change to use the normal logger now, will change to go-log in a follow-up PR.